tests/inject/dynamic/shadow-dom.tests.ts

```diff
@@ -88,4 +88,53 @@ describe('Handle Shadow-DOM', () => {
         expect(getComputedStyle(shadowRoot.querySelector('p')).color).toBe('rgb(255, 26, 26)');
     });
 
+    it('should handle defined elements', async () => {
+        container.innerHTML = multiline(
+            '<svg-container>',
+            '   <svg viewBox="0 0 100 30">',
+            '       <style class="testcase-style">#svg-text { color: red; }</style>',
+            '       <text id="svg-text" y="20">I am SVG</text>',
+            '   </svg>',
+            '</svg-container>',
+        );
+        class SVGContainer extends HTMLElement {
+            constructor() {
+                super();
+                const shadowRoot = this.attachShadow({mode: 'open'});
+                shadowRoot.append(...this.childNodes as any);
+            }
+        }
+        customElements.define('svg-container', SVGContainer);
+
+        createOrUpdateDynamicTheme(theme, null, false);
+
+        await timeout(100);
+        expect(getComputedStyle(document.querySelector('svg-container').shadowRoot.querySelector('text')).color).toBe('rgb(255, 26, 26)');
+    });
+
+    it('should react to defined elements', async () => {
+        container.innerHTML = multiline(
+            '<svg-container2>',
+            '   <svg viewBox="0 0 100 30">',
+            '       <style class="testcase-style">#svg-text { color: red; }</style>',
+            '       <text id="svg-text" y="20">I am SVG</text>',
+            '   </svg>',
+            '</svg-container2>',
+        );
+        class SVGContainer extends HTMLElement {
+            constructor() {
+                super();
+                const shadowRoot = this.attachShadow({mode: 'open'});
+                shadowRoot.append(...this.childNodes as any);
+            }
+        }
+
+        createOrUpdateDynamicTheme(theme, null, false);
+
+        await timeout(100);
+        customElements.define('svg-container2', SVGContainer);
+        await timeout(100);
+        expect(getComputedStyle(document.querySelector('svg-container2').shadowRoot.querySelector('text')).color).toBe('rgb(255, 26, 26)');
```

Comment:
If the test doesn't work, you first need to determine why, and if nothing helps remove it.

By the way, each test is independent and you can remove `-2` postfixes from class names etc.

Reply:
> By the way, each test is independent

Wrong, it is on the same page and the CustomElementRegistry isn't cleared therefore obviously so removing `-2`.
```
NotSupportedError: Failed to execute 'define' on 'CustomElementRegistry': the name "svg-container" has already been used with this registry
```